windows-agent/internal/daemon/daemontestutils/networking_mock.go

```diff
@@ -144,7 +144,7 @@ func getLocalPrivateIP() net.IP {
 		return nil
 	}
 	for _, addr := range addrs {
-		if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() && ipnet.IP.IsPrivate() {
+		if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() && ipnet.IP.IsPrivate() && ipnet.IP.To4() != nil {
```

Comment:
Very good finding! I'd update the function name to `getLocalPrivateIPv4` (and the comment above it) to reflect that requirement. After all the bug existed because that requirement wasn't explicit.